### PR TITLE
Store

### PR DESCRIPTION
**Pull request by Killian-pro**
@paour

### CODE REVIEW

File: src/api/callApi.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  });
+  return response.data;
+};
+
+export const getChapter = async (bookId: number) => {
+  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
+    query:
+      'query chapters($bookId:Int){viewer{chapters(bookIds:[$bookId]){hits{id title url valid}}}}',
+    variables: {
+      bookId: bookId,
```
Comment (paour) [inline src/api/callApi.ts:16]: Shorthand:

```suggestion
      bookId,
```

File: src/api/callApi.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+export const getBook = async () => {
+  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
+    query:
+      'query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}',
+  });
+  return response.data;
+};
+
+export const getChapter = async (bookId: number) => {
```
Comment (paour) [inline src/api/callApi.ts:11]: Could you type the return value?

File: src/api/callApi.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
```
Comment (paour) [inline src/api/callApi.ts:12]: Is `let` useful here? (rather than `const`)?

File: src/components/CardChapter.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          source={require('../assets/noImage.png')}
+        />
+      ) : (
+        <Image source={{uri: chapter?.url}} width={100} height={100} />
+      )}
+      <Text style={styles.textColor}> {chapter.title}</Text>
+      <TouchableOpacity
+        onPress={() => {
+          useAppStore.setState({chapterShow: [...chapterShow, chapter.id]});
+          setRefresh(!refresh);
```
Comment (paour) [inline src/components/CardChapter.tsx:25]: Is there another way for this widget to refresh when it's clicked (without the explicit refresh mechanism)?

File: src/components/CardChapter.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          style={{width: 100, height: 100}}
+          source={require('../assets/noImage.png')}
+        />
+      ) : (
+        <Image source={{uri: chapter?.url}} width={100} height={100} />
+      )}
+      <Text style={styles.textColor}> {chapter.title}</Text>
+      <TouchableOpacity
+        onPress={() => {
+          useAppStore.setState({chapterShow: [...chapterShow, chapter.id]});
```
Comment (paour) [inline src/components/CardChapter.tsx:24]: What happens if I click this twice?

File: src/screens/BookByFilterScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    });
+  }, [props.route.params.filterName]);
+
+  return (
+    <SafeAreaView style={{flex: 1}}>
+      <HeaderDefault title={props.route.params.filterName} />
+      <TouchableOpacity
+        style={{alignSelf: 'center', margin: 15}}
+        onPress={() => {
+          setExamMode(!examMode);
```
Comment (paour) [inline src/screens/BookByFilterScreen.tsx:48]: Using the callback version of the state setter is more robust.

```suggestion
          setExamMode(prevExamMode => !prevExamMode);
```

File: src/screens/BookByFilterScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        </Text>
+      </TouchableOpacity>
+
+      <FlatList
+        style={styles.body}
+        data={
+          examMode
+            ? books.filter(
+                book =>
+                  book?.levels?.[0].name.startsWith('Terminale') ||
```
Comment (paour) [inline src/screens/BookByFilterScreen.tsx:61]: Would this find a book where the levels are `["3e", "2nd"]`?

File: src/screens/BookByFilterScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            return (
+              (levels &&
+                levels.some((level: any) => level.name === filterName)) ||
+              (subjects && subjects.some((sub: any) => sub.name === filterName))
+            );
+          })
+          .sort((a: bookType, b: bookType) => {
+            const aLevels = a.levels || [];
+            const bLevels = b.levels || [];
+            const aLevelName = aLevels.length > 0 ? aLevels[0].name : '';
```
Comment (paour) [inline src/screens/BookByFilterScreen.tsx:34]: Use optional chaining

```suggestion
            const aLevelName = a.levels?.[0]?.name ?? '';
```

File: src/api/callApi.ts (hunk 2 above)
Comment (Killian-pro) [reply src/api/callApi.ts:11]:
```
interface BookApiResponse {
  data: {
    viewer: {
      books: {
        hits: bookType[];
      };
    };
  };
}
export const getBook = async (): Promise<BookApiResponse> => {
  let response = await axios.post(`https://api-dev.lelivrescolaire.fr/graph`, {
    query:
      'query{viewer{books{hits{id displayTitle url subjects{name}levels{name}valid}}}}',
  });
  return response.data;
};
```

Comment (paour) [reply src/api/callApi.ts:11]: Cool. Now, can you create a generic type that can handle both `books` and `chapters`?

Comment (Killian-pro) [reply src/api/callApi.ts:11]: I create a new branch with fix.
I have left store but => AxiosError: Request failed with status code 503

Comment (paour) [reply src/api/callApi.ts:11]: If you get API errors, it's useful to capture a `curl` that can be used as a reproduction of the error. Many factors can lead to an API error, and without knowing the request that was sent, it's hard to infer what went wrong.

The two sample requests do return the expected data.

**Pull request merged**